### PR TITLE
chore: playwright tasks

### PR DESCRIPTION
**Pull request by reenphygeorge**
### Changes Made:

* Login setup
* User Registration test case
* Todo related test cases
* Comments related test cases

### CODE REVIEW

File: playwright-tests/.env [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# .env.development
+
+NEETO_CI_JOB_ID=ABCD123
+RECORD_KEY=ABCD123
+PROJECT_ID=ABCD123
+TAG=development
```
Comment (VarunSriram99) [inline playwright-tests/.env]: This is just an example. You can avoid this.

File: playwright-tests/e2e/pom/tasks.ts [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        await starIcon.click();
+        await expect(starIcon).toHaveClass(/ri-star-fill/i);
+        await expect(
+            this.page.getByTestId("tasks-pending-table").getByRole("row").nth(1)
+        ).toContainText(taskName);
+    };
+
+    createCommentAndVerify = async ({ comment, taskName }: Comments) => {
+        const addCommentApi = this.page.waitForResponse(response =>
+            response.url().includes(taskName.split(" ").join("-"))
```
Comment (VarunSriram99) [inline playwright-tests/e2e/pom/tasks.ts]: Please create a new util called hyphenize for cases like this.
Have you handled a case where the task title can have mixed cases and special characters like @, # etc?

**Review by VarunSriram99 — CHANGES_REQUESTED** [review]
@reenphygeorge This doesn't cover all the comments related test cases. Please go through the list and make sure you've covered all them.
Please post a screenshot of the test results after running them as well.

File: playwright-tests/e2e/pom/tasks.ts (hunk 2 above)
Comment (reenphygeorge) [reply playwright-tests/e2e/pom/tasks.ts]: @VarunSriram99  Comments can have any type of characters right? So I didn't got what cases to be handled there.

**Comment by reenphygeorge** [comment]
IMG-20240227-WA0012-01

File: playwright-tests/e2e/pom/tasks.ts (hunk 2 above)
Comment (reenphygeorge) [reply playwright-tests/e2e/pom/tasks.ts]: @VarunSriram99 can you review this.

File: playwright-tests/e2e/utils/hyphenize.ts [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export const hyphenize = (str: string) => {
+    return str.split(" ").join("-");
+};
```
Comment (VarunSriram99) [inline playwright-tests/e2e/utils/hyphenize.ts:3]: An arrow function returns by default if not wrapped in a block.
```suggestion
export const hyphenize = (str: string) => str.split(" ").join("-");
```

File: playwright-tests/e2e/tests/comments.spec.ts [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+        await page.goto("/");
+
+        await test.step("step 1 - Create a new task and assign it to other user", () =>
+            taskPage.createTaskAndVerify({
+                taskName: todoName,
+                username: "Sam Smith",
+            }));
+
+        await test.step("step 2 - Go to the previously created task", async () => {
+            // const endPoint = todoName.replace(/\s+/g, "-");
```
Comment (VarunSriram99) [inline playwright-tests/e2e/tests/comments.spec.ts:35]: Please remove unnecessary comments.

File: playwright-tests/e2e/pom/tasks.ts [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+        await expect(starIcon).toHaveClass(/ri-star-fill/i);
+        await expect(
+            this.page.getByTestId("tasks-pending-table").getByRole("row").nth(1)
+        ).toContainText(taskName);
+    };
+
+    createCommentAndVerify = async ({ comment, taskName }: Comments) => {
+        const addCommentApi = this.page.waitForResponse(response =>
+            response.url().includes(hyphenize(taskName))
+        );
```
Comment (VarunSriram99) [inline playwright-tests/e2e/pom/tasks.ts:91]: @reenphygeorge create a new task titled `hello#@Oliver`. Check if this test passes.

**Review by VarunSriram99 — CHANGES_REQUESTED** [review]
@reenphygeorge Some edge cases still haven't been covered.

**Comment by VarunSriram99** [comment]
@reenphygeorge Please take care of all the review suggestions seriously and make sure you have completed them all before you request for a review again.

**Comment by VarunSriram99** [comment]
@reenphygeorge I didn't get any updates or review requests from your end. Have you made any progress with the comments that were added?
